### PR TITLE
feat: resource contracts

### PR DESCRIPTION
**Pull request by jyecusch**
Enables clients to declare their resource requirements

### CODE REVIEW

File: contracts/proto/resource/v1/resource.proto [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+option java_outer_classname = "Resources";
+option php_namespace = "Nitric\\Proto\\Resource\\V1";
+option csharp_namespace = "Nitric.Proto.Resource.v1";
+
+// Nitric Resource Service
+// The service definition exists to allow a nitric application to declare peripheral dependencies
+service ResourceService {
+  // Declare a resource for the nitric application
+	// At Deploy time this will create resources as part of the nitric stacks dependency graph
+	// At runtime
```
Comment (tjholm) [inline contracts/proto/resource/v1/resource.proto:19]: Need to fix formatting here

**Review by tjholm — CHANGES_REQUESTED** [review]
Just a formatting fix required as far as I can see

File: contracts/proto/resource/v1/resource.proto [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  // Collection Permissions: 4XX
+  CollectionDocumentRead = 400;
+  CollectionDocumentWrite = 401;
+  CollectionDocumentDelete = 402;
+  CollectionQuery = 403;
+  CollectionList = 404;
+}
+
+// Result from putting the secret to a Secret Store
```
Comment (tjholm) [inline contracts/proto/resource/v1/resource.proto:89]: copy/paste error comment here.

**Review by tjholm — CHANGES_REQUESTED** [review]

**Review by davemooreuws — APPROVED** [review]
Looks good once Tim's requested changes are done.

File: contracts/proto/resource/v1/resource.proto (hunk 1 above)
Comment (tjholm) [reply contracts/proto/resource/v1/resource.proto]: fixed.

File: contracts/proto/resource/v1/resource.proto (hunk 2 above)
Comment (tjholm) [reply contracts/proto/resource/v1/resource.proto]: fixed.

**Review by tjholm — APPROVED** [review]

**Pull request merged**